### ISSUE TITLE
Understanding exit codes and their meaning(s)

### ISSUE THREAD

**Issue by whoisj** [git-lfs/git-lfs#2613]
Does Git-lfs use exit code to hint at outcomes? Example: if Git exits with a 128 we know it encountered a fatal error, where as a 129 indicates a usage error, etc.

I'm seeing git-lfs exit with a 2, and I'm assuming exit_core != 0 is a failure. Are there any documents regarding exit codes and their meaning?

**Comment by technoweenie** [comment]
Nope. I'd be open to defining meaningful exit codes and documenting it somewhere. It's not something I have strong opinions about.

**Comment by technoweenie** [comment]
0 is definitely the only success exit code that LFS returns.

**Comment by trejkaz** [comment]
If we're asserting that non-zero definitely means an error, does this mean that there is some edge case where `git lfs` can fail without printing an error message?

Output from our Jenkins build shows it executing `git lfs` but then deciding it's a failure even though there isn't any sign of an error...

    [ubuntu]  > git lfs pull origin # timeout=30
    [ubuntu] Failed in branch ubuntu

**Comment by technoweenie** [comment]
I hope not. Something should be sent to STDERR at least. If this is a repeatable error, changing it to run `GIT_TRACE=1 git lfs pull origin` will spew debug details to STDOUT.

**Comment by trejkaz** [comment]
That's worth a shot, I guess. It should at least help confirm whether Jenkins itself is hiding output of errors... it's the prime suspect for our issue right now and I already reported it on their tracker anyway, a day before I started wondering about the behaviour of git-lfs itself.
